### PR TITLE
replace mako templates for tool lineage and quotas grids

### PR DESCRIPTION
**Pull request by guerler**
Removes grid and form makos used for the admin grid quota and tool version handling. Branched from #4405.

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/controllers/admin.py [hunk 1]
```diff
@@ -549,246 +540,203 @@ def users_list( self, trans, **kwd ):
...
+                            'name'    : 'amount',
+                            'label'   : 'Amount',
+                            'help'    : 'Examples: "10000MB", "99 gb", "0.2T", "unlimited"'
+                        }, {
+                            'name'    : 'operation',
+                            'label'   : 'Assign, increase by amount, or decrease by amount?',
+                            'options' : [ ('=', '=' ), ( '+', '+' ), ( '-', '-' ) ]
+                        }, {
+                            'name'    : 'default',
+                            'label'   : 'Assign, increase by amount, or decrease by amount?',
```
Comment (martenson) [inline lib/galaxy/webapps/galaxy/controllers/admin.py]: this should probably read `Is this quota a default for a class of users (if yes, what type)?`

**Review by martenson — CHANGES_REQUESTED** [review]
when trying to create a quota I get this
```
DataError: (psycopg2.DataError) invalid input syntax for integer: "8ad7aaf8de1bd67f"
LINE 3: WHERE galaxy_user.id = '8ad7aaf8de1bd67f'
```

Also when creating quota please re-label the button to something other than `save settings` - e.g. `Create quota`.

**Review by martenson — CHANGES_REQUESTED** [review]
when trying to search on the quota grid I get an alert with `Grid refresh failed` and this stacktrace:
<detail>
```
galaxy.web.framework.decorators ERROR 2017-08-16 14:17:24,596 Uncaught exception in exposed API method:
Traceback (most recent call last):
  File "/Users/marten/devel/git/galaxy/lib/galaxy/web/framework/decorators.py", line 154, in decorator
    rval = func( self, trans, *args, **kwargs)
  File "/Users/marten/devel/git/galaxy/lib/galaxy/web/framework/decorators.py", line 98, in decorator
    return func( self, trans, *args, **kwargs )
  File "/Users/marten/devel/git/galaxy/lib/galaxy/webapps/galaxy/controllers/admin.py", line 574, in quotas_list
    return self.quota_list_grid( trans, **kwargs )
  File "/Users/marten/devel/git/galaxy/lib/galaxy/web/framework/helpers/grids.py", line 221, in __call__
    total_num_rows = query.count()
  File "/Users/marten/devel/git/galaxy/.venv/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 2980, in count
    return self.from_self(col).scalar()
  File "/Users/marten/devel/git/galaxy/.venv/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 2749, in scalar
    ret = self.one()
  File "/Users/marten/devel/git/galaxy/.venv/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 2718, in one
    ret = list(self)
  File "/Users/marten/devel/git/galaxy/.venv/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 2761, in __iter__
    return self._execute_and_instances(context)
  File "/Users/marten/devel/git/galaxy/.venv/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 2776, in _execute_and_instances
    result = conn.execute(querycontext.statement, self._params)
  File "/Users/marten/devel/git/galaxy/.venv/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 914, in execute
    return meth(self, multiparams, params)
  File "/Users/marten/devel/git/galaxy/.venv/lib/python2.7/site-packages/sqlalchemy/sql/elements.py", line 323, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/Users/marten/devel/git/galaxy/.venv/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1010, in _execute_clauseelement
    compiled_sql, distilled_params
  File "/Users/marten/devel/git/galaxy/.venv/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1146, in _execute_context
    context)
  File "/Users/marten/devel/git/galaxy/.venv/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1341, in _handle_dbapi_exception
    exc_info
  File "/Users/marten/devel/git/galaxy/.venv/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 202, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/Users/marten/devel/git/galaxy/.venv/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1139, in _execute_context
    context)
  File "/Users/marten/devel/git/galaxy/.venv/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 450, in do_execute
    cursor.execute(statement, parameters)
ProgrammingError: (psycopg2.ProgrammingError) can't adapt type 'property' [SQL: 'SELECT count(*) AS count_1 \nFROM (SELECT quota.id AS quota_id, quota.create_time AS quota_create_time, quota.update_time AS quota_update_time, quota.name AS quota_name, quota.description AS quota_description, quota.bytes AS quota_bytes, quota.operation AS quota_operation, quota.deleted AS quota_deleted \nFROM quota \nWHERE lower(quota.name) LIKE %(lower_1)s OR lower(quota.description) LIKE %(lower_2)s OR lower(%(lower_3)s) LIKE %(lower_4)s ORDER BY lower(quota.name) ASC) AS anon_1'] [parameters: {'lower_1': u'%test%', 'lower_3': <property object at 0x106521d08>, 'lower_2': u'%test%', 'lower_4': u'%test%'}]
```
</detail>

**Comment by martenson** [comment]
Also when switching between grids the search terms seem to stick. Also multiple same searchterms are allowed. 
screenshot 2017-08-16 14 19 35

**Comment by martenson** [comment]
So the grid search error is also happening on Main...

**Comment by guerler** [comment]
"when switching between grids the search terms seem to stick" is an issue I am aware of and a related to how we handle url parameters i.e. passing them through the backend. It is something we can fix when working on the router again.

**Comment by martenson** [comment]
When using the advanced search, after triggering, the advanced search switches to the normal search. The previous behavior was that it stayed in the advanced setting after delivering the results.

**Review by martenson — APPROVED** [review]

**Comment by martenson** [comment]
So there are few items left after merging:

- When using the advanced search, after triggering, the advanced search switches to the normal search. The previous behavior was that it stayed in the advanced setting after delivering the results.
- When switching between grids the search terms seem to stick. 
- Multiple same searchterms are allowed.

**Comment by guerler** [comment]
Thanks a lot for the detailed review.

**Pull request merged**